spring-boot/admin/src/main/resources/templates/admin/fragments/wizard/wizard.html

```diff
@@ -1,348 +1,381 @@
-                    }
-                return true;
-
+                autoAdjustHeight: true,
+                keyNavigation: false,
+                onLeaveStep: leaveAStepCallback,
+                onShowStep: onShowStepEvent,
+                onFinish: onFinishCallback,
             }
-
-            function onFinishCallback(objs, context) {
-
-                if ($("#serviceName").val().trim() == "")
-                {
-                    alert("You must fill service name");
-                }
-                if ($("#serviceUrl").val().trim() == "")
-                {
-                    alert("You must fill service url");
-                }
-
-                if (!isValidURL($("#serviceUrl").val().trim()))
-                {
-                    alert("Service URL must be valid and start with http:// or https://");
-                }
-
-                if ($("#serviceForFree").val() == "")
-                {
-                    alert("You must select either free service or not");
-                } else
-                {
-                    if ($("#serviceForFree").val() == "false" && $("#monthlyChargeInput").val() == "")
-                    {
-                        alert("You must select either monthly charge or charge per call");
-                    }
-
-                    if ($("#serviceForFree").val() == "false" && $("#serviceCost").val() == "")
-                    {
-                        alert("You must set up a cost for your service");
-                    }
-                }
-
-                if ($("#serviceName").val().trim() != "" && $("#serviceUrl").val().trim() && isValidURL($("#serviceUrl").val().trim()))
-                {
-                    if ($("#serviceForFree").val() !== "")
-                    {
-                        if ($("#serviceForFree").val() == "true")
-                        {
-                            $("#wizardForm").submit();
-                            $('#wizard').hide();
-                            $('#busy_indicator').show();
-                        } else
-                        {
-                            if ($("#monthlyChargeInput").val() !== "" && $("#serviceCost").val() !== "")
-                            {
-                                $("#wizardForm").submit();
-                                $('#wizard').hide();
-                                $('#busy_indicator').show();
-                            }
-                        }
-                    }
-
-                }
+        );
+
+        $('.buttonNext').addClass('btn btn-success');
+        $('.buttonPrevious').addClass('btn btn-primary');
+        $('.buttonFinish').addClass('btn btn-default');
+
+    }
+
+    function leaveAStepCallback(obj, context) {
+        if (context.toStep > context.fromStep)
+            return validateSteps(context.fromStep);
+        else
+            return true;
+    }
+
+    function validateSteps(stepnumber) {
```

Comment:
Depending on the step number, this should only call the relevant validation methods.

Reply:
What do you mean?

Reply:
Why does the method needs the stepnumber, when actually all the steps are checked?

Reply:
I assume he means something like 
```
switch (stepnumber):
    2: 
        return validateServiceName();
    3: 
        return validateURL();
etc.
```
There is no need to check if step 4 is valid when you are only in step 2. And if you are in step 4 there is no need to check step 2 as it has already been verified to be ok.

Reply:
I just had an idea about why the structure is like this. Imagine a user is in step 3 (stepnumber is 3) but wants to change his entered value in step 2. So he goes back and applies the change. I think the stepnumber is still 3 because its the counter variable for the latest active step and can't decrease. This is why you always have to check all lower stepnumbers. 

However, I think this is a very minor problem and we should not invest too much effort, as the code is working. Imo, we can keep it as it is.

Reply:
Ok, so I checked how it worked and actually is not a counter, in there I am using the context.fromStep (this is a predefined function from the SmartWizard plugin which contains the steps from and to).
I will still try to solve this issue

Reply:
Solved, now a switch case is used and each validation is only checked for is specific step